### ISSUE TITLE
Crash when more than 11 devices

### ISSUE THREAD

**Issue by tvixen** [kakopappa/arduino-esp8266-alexa-multiple-wemo-switch#6]
Hi there
And thanks for a great code .... Compilation was working 1.st time with the onboard diodes.

One question ...... When i compile with 11 devices it goes ok every time....
(heres the connect log)

Connecting to WiFi
Connecting ......
Connected to WifiTestNet
IP address: 10.0.0.182
RSSI:-77
Begin multicast ..
Udp multicast server started at 239.255.255.250:1900
WebServer started on port: 80
WebServer started on port: 81
WebServer started on port: 82
WebServer started on port: 83
WebServer started on port: 84
WebServer started on port: 87
WebServer started on port: 88
WebServer started on port: 89
WebServer started on port: 90
WebServer started on port: 94
WebServer started on port: 95
Adding switches upnp broadcast responder
Adding switch : Samsung T.V. index : 0
Adding switch : T.V. Mute index : 1
Adding switch : T.V. Channel index : 2
Adding switch : T.V. sound index : 3
Adding switch : T.V. one index : 4
Adding switch : Amplifier index : 5
Adding switch : Amplifier sound index : 6
Adding switch : Amplifier aux index : 7
Adding switch : Amplifier tuner index : 8
Adding switch : Amplifier mute index : 9
Adding switch : VacuumCleaner index : 10

BUT
When I compile with 12 devices the witty esp8266 reboot every 10 second. 
It connect to wifi... and the reboot after 5 sec.

When i compile with 13 devices the witty esp8266 won't connect to wifi 
and roboot every 5 second.

Same goes with devices up to 14 and over. (I could really use 20)
(I have a 100uf over the 3.3V supply for stability)

**Have you seen this issue before, or have any idea of what is going wrong ?**
Thanks in advance.

**Comment by kakopappa** [comment]
that's a lot of switches... I have not tried using more than 3 or 4. 

have you looked at the crash log? 
https://github.com/me-no-dev/EspExceptionDecoder

**Comment by tvixen** [comment]
Ok...seems like a great tool. I will try it out and see if I can track it..... thanks.

**Comment by laz2wiringpi** [comment]
i have found a similar issue ... 
random works for 4  or rarely 5 devices 

when i added a define DEBUG to switch.h 
and  
        #if defined(DEBUG)
                   Serial.print*(" XXXX");   
         #endif 
 to remove serial.* code ...

the issue so far  seems to work  ...  and i can add 6 devices 


is suspect its the time for processing serial data and or out of  memory.
 

and it would seem there are a lot of flavors of the esp boards .  
  Adafruit HUZZAH ESP8266 breakout.

**Comment by tvixen** [comment]
Yes there's a lots of brands. Mine is the Witty ESP8266-12F  board.
Did you try without your laptop. Just with a powersupply around 2A.

**Comment by laz2wiringpi** [comment]
Not power issue I use a lab power supply.The issue is the server loop in main Each Web server can't respond while handling another request  Modified the payload to min bare bones Setup.xml without debug serial works 
Best just to use more than one device past 5 to 6 ports 



Sent via the Samsung Galaxy S7, an AT&T 4G LTE smartphone
-------- Original message --------From: Tvixen <notifications@github.com> Date: 1/16/17  11:36 AM  (GMT-05:00) To: kakopappa/arduino-esp8266-alexa-multiple-wemo-switch <arduino-esp8266-alexa-multiple-wemo-switch@noreply.github.com> Cc: laz2wiringpi <arotton@gmail.com>, Comment <comment@noreply.github.com> Subject: Re: [kakopappa/arduino-esp8266-alexa-multiple-wemo-switch] Crash when
  more than 11 devices (#6) 
Yes there's a lots of brands. Mine is the Witty ESP8266-12F  board.

Did you try without your laptop. Just with a powersupply around 2A.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/kakopappa/arduino-esp8266-alexa-multiple-wemo-switch","title":"kakopappa/arduino-esp8266-alexa-multiple-wemo-switch","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/kakopappa/arduino-esp8266-alexa-multiple-wemo-switch"}},"updates":{"snippets":[{"icon":"PERSON","message":"@tvixen in #6: Yes there's a lots of brands. Mine is the Witty ESP8266-12F  board.\r\nDid you try without your laptop. Just with a powersupply around 2A. "}],"action":{"name":"View Issue","url":"https://github.com/kakopappa/arduino-esp8266-alexa-multiple-wemo-switch/issues/6#issuecomment-272909031"}}}

**Comment by tvixen** [comment]
Well it shouldn't be necessary to do two things at the same time. as Alexa can only call call one webserver port at the time. 
1. Ask Alexa
2. Alexa send the request to port 80
3. ESP8266, Do the request
4. ESP8266 go back to listen
Repeat------
2. Alexa send the request to port 81
3. ESP8266, Do the request
4. ESP8266 go back to listen

**Comment by laz2wiringpi** [comment]
It's the discovery ... setup.xml

I think it polls every so often not positive though

**Comment by tvixen** [comment]
Remember to delete Alexa devices in your app, before you scan for new devices.  Remember to ask Alexa, and do NOT scan via the app.
